multiqc/modules/odgi/odgi.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from odgi stats """
+
+from __future__ import print_function
+import logging
+from collections import OrderedDict
+
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.plots import linegraph
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    The MultiQC module to parse and plot odgi stats output.
+    """
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='Odgi', anchor='odgi',
+                                            href='https://github.com/vgteam/odgi',
+                                            info='is an optimized dynamic graph/genome implementation.')
+
+        self.odgi_stats_map = dict()
+
+        # Find and load any odgi stats file
+        log_files = self.find_log_files('odgi')
+
+        # Parse odgi stats data
+        for f in log_files:
+            self.parse_odgi_stats_report(f)
+            self.add_data_source(f)
+
+        self.odgi_stats_map = self.ignore_samples(self.odgi_stats_map)
+
+        if not self.odgi_stats_map:
+            raise UserWarning
+
+        log.info('Found {} reports'.format(len(self.odgi_stats_map)))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.odgi_stats_map, 'multiqc_odgi_stats')
+
+        # Add the general section containing the general odgi stats
+        self.plot_general_odgi_stats()
+
+        # Plot the odgi stats metrics as a lineplot
+        self.plot_odgi_metrics()
+
+    def parse_odgi_stats_report(self, f):
+        """
+        The stats reports assume the following structure:
+
+        length    nodes    edges    paths
+        8778    168    243    35
+        #mean_links_length
+        path    in_node_space    in_nucleotide_space    num_links_considered
+        all_paths    9.75053    497.321    942
+        #sum_of_path_node_distances
+        path    in_node_space    in_nucleotide_space    nodes    nucleotides    num_penalties    num_penalties_different_orientation
+        all_paths    20.0686     19.5609    977     51365     90    0
+        """
+        lines = f['f'].splitlines()
+        # parse the generals stats
+        stats = lines[1].split('\t')
+        # parse the mean links length
+        mean_links_length = lines[4].split('\t')
+        # parse sum of path nodes distances
+        sum_of_path_nodes_distances = lines[7].split('\t')
```

Comment:
I generally try to avoid addressing specific line numbers if at all possible. It is surprisingly common for things like HPC schedulers etc to insert unexpected lines into the top of files, which can offset everything and cause chaos.

If possible, it would give more robust code to parse by line contents instead.